pkg/server/autoconfig/auto_config.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package autoconfig
```

Comment:
depending on how the package structure comes together, you may want to move this into a subpackage called autoconfigjob or something. The jobs end up taking on a lot of dependencies. If there are other APIs you're going to want lower dependency packages to use, you'll be sad having the job in the package at the root of the subsystem.